dags/common/immersion_facilitee/helpers.py

```diff
@@ -0,0 +1,133 @@
+import datetime
+import logging
+import re
+import time
+from typing import Dict, List
+
+import httpx
+import pandas as pd
+from airflow.models import Variable
+from dateutil.relativedelta import relativedelta
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+from sqlalchemy.orm import Session
+
+from dags.common import db
+from dags.common.anonymize_sensible_data import hash_content, normalize_sensible_data
+from dags.common.immersion_facilitee.models import Conventions
+
+
+logger = logging.getLogger(__name__)
+
+
+def api_client() -> httpx.Client:
+    """
+    Crée un client HTTP pour l'API Immersion Facilitee
+    :return: un client HTTP configuré
+    """
+    return httpx.Client(
+        base_url=Variable.get("API_IMMERSION_FACILITEE_BASE_URL"),
+        headers={"authorization": str(Variable.get("API_IMMERSION_FACILITEE_TOKEN"))},
+        timeout=httpx.Timeout(timeout=5, read=30),
+    )
+
+
+def get_all_items(path: str) -> List[Dict]:
+    """
+    Récupère toutes les conventions des dernières 5 ans
+    :param path: chemin de l'API
+    :return: liste de dictionnaires contenant les conventions
+    """
+
+    # Aujourd'hui on boucle sur les jours car jusqu'à maintenant pour les SIAE il n'y a pas 100
+    # ou plus conventions. L'API IF évoluera probablement permettant la pagination et une date de mise à jour
+
+    start_date_less_or_equal = datetime.date.today()
+    start_date_greater = start_date_less_or_equal - relativedelta(years=5)
+
+    client = api_client()
+
+    data = []
+
+    for date_temp in pd.date_range(start_date_greater, start_date_less_or_equal, freq="D"):
+        str_date_g = date_temp
+        str_date_l = date_temp + relativedelta(days=1)
+        response = client.get(
+            path,
+            params={
+                "withStatuses[]": ["ACCEPTED_BY_VALIDATOR"],
+                "startDateLessOrEqual": str_date_l.strftime("%Y-%m-%d"),
+                "startDateGreater": str_date_g.strftime("%Y-%m-%d"),
+            },
+        )
+        response.raise_for_status()
+
+        data_partial = response.json()
```

Comment:
Vu qu'on est désormais à jour, tu peux _chain_ les appels :
```suggestion
        data_partial = response.raise_for_status().json()
```